arXiv.org.js

```diff
@@ -35,217 +35,396 @@
-		dc: 'http://purl.org/dc/elements/1.1/',
-		xsi: 'http://www.w3.org/2001/XMLSchema-instance',
-		n: 'http://www.openarchives.org/OAI/2.0/' // Default
-	};
-	var hasPreprint;
-	if (ZU.fieldIsValidForType('title', 'preprint')) {
-		hasPreprint = true;
-	}
-	var newItem;
-	if (hasPreprint) {
-		newItem = new Zotero.Item("preprint");
-	}
-	else {
-		newItem = new Zotero.Item("report");
-	}
-	var xml = (new DOMParser()).parseFromString(text, "text/xml");
-	var dcMeta = ZU.xpath(xml, '//n:GetRecord/n:record/n:metadata/oai_dc:dc', ns)[0];
-
-	newItem.title = getXPathNodeTrimmed(dcMeta, "dc:title", ns);
-	getCreatorNodes(dcMeta, "dc:creator", newItem, "author", ns);
-	var dates = ZU.xpath(dcMeta, './dc:date', ns)
-		.map(element => element.textContent)
-		.sort();
-	if (dates.length > 0) {
-		if (version && version < dates.length) {
-			newItem.date = dates[version - 1];
-		}
-		else {
-			// take the latest date
-			newItem.date = dates[dates.length - 1];
-		}
-	}
-	
-	
-	var descriptions = ZU.xpath(dcMeta, "./dc:description", ns);
-	
-	// Put the first description into abstract, all other into notes.
-	if (descriptions.length > 0) {
-		newItem.abstractNote = ZU.trimInternal(descriptions[0].textContent);
-		for (let j = 1; j < descriptions.length; j++) {
-			var noteStr = ZU.trimInternal(descriptions[j].textContent);
-			newItem.notes.push({ note: noteStr });
-		}
-	}
-	var subjects = ZU.xpath(dcMeta, "./dc:subject", ns);
-	for (let j = 0; j < subjects.length; j++) {
-		var subject = ZU.trimInternal(subjects[j].textContent);
-		newItem.tags.push(subject);
+function parseSingleEntry(entry) {
+	let newItem = new Zotero.Item("preprint");
+
+	newItem.title = ZU.trimInternal(text(entry, "title"));
+	newItem.date = ZU.strToISO(text(entry, "updated"));
+	entry.querySelectorAll(`author > name`).forEach(node => newItem.creators.push(ZU.cleanAuthor(node.textContent, 'author', false)));
+
+	newItem.abstractNote = ZU.trimInternal(text(entry, "summary"));
+
+	let comments = entry.querySelectorAll("comment");
+
+	for (let comment of comments) {
+		let noteStr = ZU.trimInternal(comment.textContent);
+		newItem.notes.push({ note: `Comment: ${noteStr}` });
 	}
-					
-	var identifiers = ZU.xpath(dcMeta, "./dc:identifier", ns);
-	for (let j = 0; j < identifiers.length; j++) {
-		var identifier = ZU.trimInternal(identifiers[j].textContent);
-		if (identifier.substr(0, 4) == "doi:") {
-			newItem.DOI = identifier.substr(4);
-		}
-		else if (identifier.substr(0, 7) == "http://") {
-			newItem.url = identifier;
-		}
+
+	let categories = Array.from(entry.querySelectorAll("category"))
+		.map(el => el.getAttribute("term"))
+		.map((sub) => {
+			let mainCat = sub.split('.')[0];
+			if (mainCat !== sub && arXivCategories[mainCat]) {
+				return arXivCategories[mainCat] + " - " + arXivCategories[sub];
+			}
+			else {
+				return arXivCategories[sub];
+			}
+		})
+		.filter(Boolean);
+	newItem.tags.push(...categories);
+
+	let arxivURL = text(entry, "id").replace(/v\d+/, '');
```

Comment:
We need a clear determination here wrt to user's expectations. The Atom endpoint always indicates the version in the url, so that's why the version is removed here. However, we need to clarify when and where the version should be saved in the item's url.